### DISCUSSION TITLE
Add Coupon fields that correspond to Subscription trial days

### DISCUSSION THREAD

**Discussion by grantnhoward** [craftcms/commerce#2316] [Ideas]
### Description
Coupons work for everything except subscriptions currently, and the only subscription gateway (Stripe) already has a mechanism for accepting coupons that correspond to discounts and free trial periods. It would be ideal to have these systems working together, or at least include a mechanism to pass the information back via Craft Discount code -> (gateway) trialDays value when processing a new subscription.

### Steps to reproduce

1. Unable to include promotional codes for free trial months of subscriptions.
2.

### Additional info

- Craft version: Craft Pro 3.0.0-RC17.1
- PHP version: 7.0.28
- Database driver & version: MySQL 5.6.27
- Plugins & versions: Craft Ccmmerce 2:  dev-develop

**Comment by andris-sevcenko** [comment]
I like this idea. Just have to deal with all things launch and post-launch first :)

**Comment by johnwbaxter** [comment]
Is there any timeframe on this one? I've got 2 commerce subscriptions projects that really need to be able to add discounts / coupons to their subscriptions.

**Comment by andris-sevcenko** [comment]
Not currently, unfortunately. As a stop-gap solution, you can use the [appropriate event](https://github.com/craftcms/commerce/blob/develop/docs/events.md#the-beforecreatesubscription-event) to modify the subscription before creating it, so it should be possible to have a custom module or plugin that addresses this.

**Comment by ldennison-bsd** [comment]
This is a highly necessary feature of Commerce/Stripe for Commerce. Stripe has supported discounts against subscriptions for a long while, and for e-commerce applications using Stripe and Commerce subscriptions, this is a big missing piece of functionality. Yes, custom code can always solve problems, but this is not an obscure request. It's a very common requirement to be able to apply a discount to an online purchase, even if it's a recurring purchase.

Considering the Commerce license is non-trivial money and includes a recurring yearly fee for the software, I'd expect this feature to be a high priority. We shouldn't have to develop custom modules (and thus increase our clients cost) to simply apply a discount (which Commerce already has) to a Stripe subscription. Quite frankly, it's hard to "sell" clients with recurring costs on switching to Commerce when it's missing such large functionality.

**Comment by jameslindeman** [comment]
Any update on the timeline for this?

**Comment by aakashverma12** [comment]
2020's gonna end, 2 years since the issue opened and I think its a mandatory feature.

**Comment by chadsten** [comment]
Has anyone developed a plugin for this that they would be willing to license?

**Comment by aakashverma12** [comment]
I'm developing a plugin for this, its working great, but its not fully ready yet.

> 
> 
> Has anyone developed a plugin for this that they would be willing to license?

**Comment by peter-tell** [comment]
Hey @aakashverma12  --

Very cool! Do you have an ETA when you're thinking of shipping the plugin?

**Comment by astuteo** [comment]
Anybody have an update on this one? The inability to discount subscriptions, especially considering the feature has existed in Stripe for years, makes for a pretty big hole as far as clients are concerned.

**Comment by grantnhoward** [comment]
This being 3 years ago, we ended up wrapping the new subscription event with logic to check for a "discount code" field value and add an associated number of days (15, 30, 60) to the free trial. The promo code doesn't cross the wire to stripe, but we were able to provide the appearance of expected functionality.

**Comment by astuteo** [comment]
@lukeholder Do you have a status update on subscription discounts/coupon codes for Commerce? We have a customer that was okay holding off on this in late 2018 based on the expectations at the time, but the subject bubbles up a couple times a year. At this point, if it's still entirely backlogged, I'd like to relay it to them so we can come up with a different solution.

**Comment by aaronlax** [comment]
Is there any update here? I have a number of clients who have been chomping at the bit for this feature and its getting difficult to continue to build workarounds.

**Reply by aaronlax** [reply]
Checking in again here @brandonkelly @lukeholder happy to support this feature however I can

**Reply by lukeholder** [reply]
> Checking in again here @brandonkelly @lukeholder happy to support this feature however I can

Hi @aaronlax 


We plan to move all subscriptions features to a new independent Stripe integration later this year. So core Commerce will not manage subscriptions from that point, and coupon management for Stripe would then be independent of the cart discount coupons. 

We just found that subscriptions are so unique to the subscription platform it doesn't make sense to try and abstract them in Commerce. 

If you want to look to add Stripe coupon management to your own module or plugin, or even a PR to the Stripe plugin we would be happy to help guide you or answer any questions.

Thanks.
